### PR TITLE
Add `wedgeinv` and `circleinv` shape to muxdemuxes

### LINKED ISSUE #679: how to use other shape for a node like triangle (for european invert/not symbol)?
* Similarly, what is the node for `tripoles/european not symbol=triangle`
* i.e. how to use other shape for a node like triangle (for european invert/not symbol)? 

I found out that:
* `Triangle[open, left]` (page 219, III.16.5.3 Ti*k*Z > Arrows > Reference: Arrow Tips > Geometric. 
* It says 'ts `An instance of a Kite with zero inset.` 
* so, I tried creating the node with both Triangle, triangle, Kite, kite, but none worked...
* `\node[kite] (B) at (A.bout)  {};` , error returned: `Package pgfkeys Error: I do not know the key '/tikz/kite' and I am going to ignore it. Perhaps you misspelled it. \node[kite]`

----
Citations:

Page 157, 4.22.7 European logic port usage
The node for using outside for `tripoles/european not symbol=ieee circle` is `ocirc` (open circle).

Page 156, 4.23 Flip-flops
a _negation_ flag ... will put an `ocirc` shape on the border of corresponding pin

Page 199, 6.1 Nodes to bipoles (and 4.12 terminal shapes)
You can use other shapes too, but at your own risk…

----
I was easily able to put a circle on the bottom left pin (`blpin`) of the muxdemux, but couldn't 

```latex
\documentclass[border=10px]{standalone}
\usepackage{circuitikz}
\begin{document}
\begin{tikzpicture}
\path (0,0) node[muxdemux, muxdemux def={NL=2,Lh=2,Rh=2}](mdemux){};
% Page 170 anchor=0
\path (mdemux.blpin 1) node[ocirc,anchor=0](){};
\end{tikzpicture}
\end{document}
```

image

### PR DESCRIPTION
**Pull request by Rmano**
Thanks to @yashpalgoyal1304
Fixes #679

image

### CODE REVIEW

File: doc/circuitikzmanual.tex [hunk 1]
```diff
@@ -7027,6 +7028,27 @@ \subsubsection{Mux-Demux anchors}
 }
 \end{lstlisting}
 
+\subsubsection{Adding European-style inversion markers}\label{sec:eurinv}
+
+Although you can add ``negation balls'' as seen for, for example, flip-flops (see section~\ref{sec:flip-flop-cust}), sometimes the Europen-style notation with the small triangle is preferred. The \texttt{euroinv} shape will nicely do. It'll scale with the \texttt{muxdemuxes} class, and the length and height can be changed with the keys \texttt{european inversion mark/width} (default \texttt{0.2}) and \texttt{height} (default \texttt{0.1}), with the same units that are used for the \texttt{external pins width} and similar keys.
```
Comment (yashpalgoyal1304) [inline doc/circuitikzmanual.tex:7033]: 1. I am forgetting if other node-style components have this or not, but how about sharing the "default anchor"?

Also, just would like this final opportunity to ponder over aptness of the name. Do you think it is correct and general enough - given the IEEE use of this triangle? `euroinv`? I think it is - the word invert is fairly genric word, applies partially well both to logic (invert the logic) as well as to polarity (reverse the polarity), and prepending "euro" gives good enough distinction i guess. But i don't have much experience.

**Review by yashpalgoyal1304 — COMMENTED** [review]
Also, just would like this final opportunity to ponder over aptness of the name. 
- Do you think it is correct and general enough - given the IEEE use of this triangle? `euroinv`? 

I think it is 
- the word invert is fairly genric word, 
- applies partially well both to logic (invert the logic) as well as to polarity (reverse the polarity), and 
- prepending "euro" gives good enough distinction i guess. 

But i don't have much experience.

File: doc/circuitikzmanual.tex (hunk 1 above)
Comment (Rmano) [reply doc/circuitikzmanual.tex:7033]: Well, the generic anchor is the one named `center` for every shape. You are right, I should add the `center` anchor to the description box, given that's in a non-standard position.
I'll add it!

**Comment by Rmano** [comment]
> * Do you think it is correct and general enough - given the IEEE use of this triangle? `euroinv`?

Hmm. The normal ball is called `notcirc`. Maybe `nottri` could be more on style, but I really don't like it...

File: doc/circuitikzmanual.tex (hunk 1 above)
Comment (yashpalgoyal1304) [reply doc/circuitikzmanual.tex:7033]: I think the default anchor here is the one at base
with default, i meant the one which's used when an anchor is not explicitly specified

**Comment by Rmano** [comment]
...and maybe the solution would be to create a `dotinv` shape that is the same as `notcirc` but scales with `muxdemuxes`.

Hmmm...

File: tex/pgfcircmultipoles.tex [hunk 2]
```diff
@@ -2160,4 +2160,53 @@
...
+%% definitions and shape for euroinv %<<<
+\ctikzset{european inversion mark/width/.initial = 0.2}
+\ctikzset{european inversion mark/height/.initial = 0.1}
+\pgfdeclareshape{euroinv}{
+    \savedmacro{\ctikzclass}{\edef\ctikzclass{muxdemuxes}}
+    \saveddimen{\scaledRlen}{\pgfmathsetlength{\pgf@x}{\ctikzvalof{\ctikzclass/scale}\pgf@circ@Rlen}}
+    \savedanchor\northeast{%
+        \pgfmathsetlength{\pgf@circ@scaled@Rlen}{\ctikzvalof{\ctikzclass/scale}\pgf@circ@Rlen}
+        \pgf@y=\ctikzvalof{european inversion mark/height}\pgf@circ@scaled@Rlen
+        \pgf@x=\ctikzvalof{european inversion mark/width}\pgf@circ@scaled@Rlen
```
Comment (yashpalgoyal1304) [inline tex/pgfcircmultipoles.tex:2172]: I think the default size is somewhat small and has to be increased... i am testing on it to see if it has to do with environment it's used in, but this is how it's looking on my document compared with the dimensions i previously used.

image

image

image

Comment (Rmano) [reply tex/pgfcircmultipoles.tex:2172]: Maybe
```
\ctikzset{european inversion mark/width/.initial = 0.3}
\ctikzset{european inversion mark/height/.initial = 0.15}
```
will do? (I don't want to change the lines you marked there --- I like the fact that the units are the same as `external pins` etc...)

Comment (yashpalgoyal1304) [reply tex/pgfcircmultipoles.tex:2172]: > I don't want to change the lines you marked there...

yeah it's okay, i just semi-randomly selected these lines... was not saying particularly... 
as i don't have much experience with pgf layer of code

Comment (yashpalgoyal1304) [reply tex/pgfcircmultipoles.tex:2172]: yeah, `0.3,0.15` is working good.... 

i keep getting hung up in editors, aaahh... that's why it takes too much time on my part

image

**Comment by yashpalgoyal1304** [comment]
i will give feedback on names tomorrow...

**Comment by Rmano** [comment]
About the default size: as it is now, it is even bigger (by default) than the one used in european logic ports:

image

...so I do not think that the default should be bigger. I like the 2:1 ratio, though, more than the one in the European ports, and it's more coherent with the IEEE standard. 

...and `wedgeinv` sound nice, no? Better indicate the form than the  half-true "European" thing.

**Comment by Rmano** [comment]
> i will give feedback on names tomorrow...

I am leaning onto `wedgeinv` more than `wedgenot`.

**Comment by Rmano** [comment]
```latex
\begin{circuitikz}[]
    \node [european nand port](N) at (0,0) {};
    \draw [red](N.bout) node[wedgeinv]{};
    \path (3,0) node[muxdemux, muxdemux def={NL=2,Lh=2,Rh=2,NB=1},
        external pins width = 0.5](mdemux){};
    \draw (mdemux.brpin 1) node[wedgeinv]{};
    \draw (mdemux.blpin 1) node[wedgeinv,anchor=apex]{};
    \draw [red] (mdemux.brpin 1) node[circleinv](l){};
    \draw [red] (mdemux.blpin 1) node[circleinv,anchor=apex]{};
    \draw (l.45) -- ++(.3,.3);
\end{circuitikz}
```

image

**Comment by Rmano** [comment]
image

**Comment by Rmano** [comment]
image

...I will squash all this before commit, anyway ;-)

**Comment by yashpalgoyal1304** [comment]
> ...and maybe the solution would be to create a `dotinv` shape that is the same as `notcirc` but scales with `muxdemuxes`.
\- https://github.com/circuitikz/circuitikz/pull/688#issuecomment-1407650470
> Circle shaped inversion marker: `circleinv`

umh, so, now there's `ocirc` `notcirc` and `circleinv` - all differing in just their association of class for their size ?

how about a way to specify the associativity with class within the node properties itself?? like:

`node[notcirc, class=muxdemuxes]`

**Comment by yashpalgoyal1304** [comment]
* yep, `wedge` seems much more appropriate than `tri`...
* `notwedge` sounds better - easily searchable when searching for `not` in the circuitikz documentation.
* `wedgenot` or `wedgeinv` - i can see the logic - to create differentiation for this size association with mdmux, but ... (continued below)

i'd again say, a better option in my opinion would be to have a single component `notwedge` - inline with `notcirc`. with option of specifying class in their node options.... rather than creating essentially duplicate components

**Comment by Rmano** [comment]
Unfortunately, the class is defined on definition, not on instantiation. I don't say it's impossible to do, but it would be a quite deep change (another one for 2.0 aka retirement version 😉).

Moreover, not only the class changes between the three object; also anchors are different as well as filling behavior. Especially `ocirc` is semantically very different too.

Hmm. Will think a bit, but for now I think I'll leave these names as they are now.
Thanks a lot for the comments and insights.

**Comment by yashpalgoyal1304** [comment]
i don't have any more concerns (i guess 😅)

File: doc/circuitikzmanual.tex (hunk 1 above)
Comment (Rmano) [reply doc/circuitikzmanual.tex]: Yes, but in any Ti*k*Z shape, the default anchor is the one named `center`. In this case, it coincides with `base`, and now it's shown in the "component box".

Comment (yashpalgoyal1304) [reply doc/circuitikzmanual.tex]: > but in any Ti_k_Z shape, the default anchor is the one named `center`. In this case, it coincides with `base`, and now it's shown in the "component box".

ohwww, ohkayyh. neat 👍

File: tex/pgfcircmultipoles.tex (hunk 2 above)
Comment (yashpalgoyal1304) [reply tex/pgfcircmultipoles.tex]: yeah, i have thought about smth similar for the tutorial. i am just clearing the backlog of issues from so long... i will soon get to writing the tutorial for #665

**Pull request merged**